flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterConfiguration.java

```diff
@@ -214,6 +220,15 @@ public MiniClusterConfiguration build() {
                     RestOptions.ADDRESS,
                     modifiedConfiguration.getString(RestOptions.ADDRESS, "localhost"));
 
+            if (useRandomPorts) {
+                if (!configuration.contains(JobManagerOptions.PORT)) {
```

Comment:
This feels weird. Why not force the options when `useRandomPorts` is enabled?

Reply:
Because we may extend the set of ports in the future but want to allow single ports to be configured.

I think it would be weirder if an explicit port configuration by the user is simply ignored.